plugins/node/opentelemetry-instrumentation-knex/test/index.test.ts

```diff
@@ -190,6 +190,275 @@ describe('Knex instrumentation', () => {
+            assert.strictEqual(
+              instrumentationSpans[1].name,
+              'insert :memory:.testTable1'
+            );
+
+            assert(last.name, 'parentSpan');
+          }
+        );
+      });
+
+      it('should correctly identify the table in join nested table with table', async () => {
+        const parentSpan = tracer.startSpan('parentSpan');
+        await context.with(
+          trace.setSpan(context.active(), parentSpan),
+          async () => {
+            await client.schema.createTable('testTable1', (table: any) => {
+              table.string('title');
+            });
+            await client.insert({ title: 'test1' }).into('testTable1');
+
+            await client.schema.createTable('testTable2', (table: any) => {
+              table.string('title');
+            });
+            await client.insert({ title: 'test2' }).into('testTable2');
+
+            const builder = client('testTable1').select('*');
+            const clone = builder.clone().clear('order');
+
+            const nestedQueryBuilder = builder.client
+              .queryBuilder()
+              .count('* AS count')
+              .from(clone.as('inner'))
+              .first();
+
+            const totalDoubleNested = await nestedQueryBuilder.client
+              .queryBuilder()
+              .from(nestedQueryBuilder.as('nested_query'))
+              .leftJoin('testTable2')
+              .first();
+            assert.deepEqual(totalDoubleNested, { title: 'test2', count: 1 });
+
+            parentSpan.end();
+
+            const instrumentationSpans = memoryExporter.getFinishedSpans();
+            const last = instrumentationSpans.pop() as any;
+            assertSpans(instrumentationSpans, [
+              {
+                statement: 'create table `testTable1` (`title` varchar(255))',
+                parentSpan,
+              },
+              {
+                op: 'insert',
+                table: 'testTable1',
+                statement: 'insert into `testTable1` (`title`) values (?)',
+                parentSpan,
+              },
+              {
+                statement: 'create table `testTable2` (`title` varchar(255))',
+                parentSpan,
+              },
+              {
+                op: 'insert',
+                table: 'testTable2',
+                statement: 'insert into `testTable2` (`title`) values (?)',
+                parentSpan,
+              },
+              {
+                op: 'first',
+                table: 'testTable1',
+                statement:
+                  'select * from (select count(*) as `count` from (se..',
+                parentSpan,
+              },
+            ]);
+            assert.strictEqual(instrumentationSpans[0].name, ':memory:');
+            assert.strictEqual(
+              instrumentationSpans[1].name,
+              'insert :memory:.testTable1'
+            );
+
+            assert(last.name, 'parentSpan');
```

Comment:
What are these asserts meant to verify in this context?

Reply:
I removed them, the first block is somewhat redundant and the second one, I personally believe it was not really relevant.